src/extras/auto-import.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Auto-import the first call to System.register 
+ */
+(function (global) {
+  var systemJSPrototype = global.System.constructor.prototype;
+  var systemInstantiate = systemJSPrototype.instantiate;
+  var systemRegister = systemJSPrototype.register;
+  var systemGetRegister = systemJSPrototype.getRegister;
+
+  // hook into System.register (but release it after first usage)
+  systemJSPrototype.register = function (deps, declare) {
+    // reset System.register for subsequent usage
+    systemJSPrototype.register = systemRegister;

```

Comment:
Let's not reset this on first call - it would be nice to support multiple scripts loaded in this way.

Rather, let's reset if `document.readyState !== 'loading'`.

Reply:
Unfortunately this reset technique does interfere with other extras.

Rather than a "reset" simply have a state variable that skips the internal routine.

Reply:
I agree that support for multiple scripts would be great. The trick would be to find the correct condition to run the custom code or not. In your second proposal, what would trigger a change in that state variable? Or do you see this as a state to be pushed to required deps, in order to not run automatically?

Also, should a user be able to choose which script files to run automatically? Or do you feel all scripts defined in the html should auto run?

Reply:
Actually we don't need anything more than -

```js
  if (document.readyState !== 'loading') return;
```

at the very top of the function. That is it.

Reply:
Done!
I did move it a bit down, right after the call to the original/previous `System.register`, in order to keep the register functionality working after loading.